### ISSUE TITLE
compatibility with SC

### ISSUE THREAD

**Issue by jahblah** [andmarti1424/sc-im#269]
It appears that a lot of SC functionality has been disabled or changed.  Is there a list of things
that no longer work with respect to SC files?

**Comment by andmarti1424** [comment]
hello. not really. have you check the doc? with what commands you are having problems with?

**Comment by jahblah** [comment]
I've looked at the doc, but not thoroughly (forgive me).  Mainly just to see how to do certain things.  Example commands that don't seem to work any more are:  "fkey", "tbl", "merge", and "write";  external function option renamed from "extfun" to "external_functions"; @ext() doesn't work as expected (see some of immae's posts);  no -W commandline option;  etc.   Those are things I used extensively with SC when building external scripts, so not having them available any longer would require major changes to some sheets in order to have them work with SCIM.

**Comment by jahblah** [comment]
hit wrong button.  oops.

**Comment by andmarti1424** [comment]
Yes, some commands were removed, some renamed. do you have this defined inside .sc files: "fkey", "tbl", "merge", "write" ????
regarding external functions, I will take a look at it since theres an open issue on this one.

take note that it was added lua scripting support. please take a look at the examples to see if they can help you..

**Comment by jahblah** [comment]
"do you have this defined inside .sc files: "fkey", "tbl", "merge", "write" ????"

I use those commands in sc files, yes.  It seems to be the only way to pass ranges to an external
script.  For example, the .sc file would have a line like:

fkey 3 = "merge \"|./foo.sh a0:c10 f0 /tmp/data\""

Then in foo.sh:

#!/bin/bash
operation () {
  ret=$(whatever) # do something with the data
  echo ${ret}
}
range=${1}
targetcell=${2}
datafile="${3}"
echo -n '' >${datafile}
echo "write \"${datafile}\" ${range}"
result=$(operation ${datafile})
echo "let ${targetcell} = ${ret}"

The downside is that you have to tap F3 to update the result when the data
change, but I don't know of any other way to do it, since you can't pass
ranges, or do back and forth communication, through @ext().

**Comment by jahblah** [comment]
fkey 3 = "merge \"|./foo.sh a0:c10 f0 /tmp/data\""
Escapes got lost when pasting above.

**Comment by reportaman** [comment]
@andmarti1424 https://github.com/n-t-roff/sc has ``set mouse``, and claims to be a fork of the latest sc, not sure if mouse was a later addition though.

**Comment by andmarti1424** [comment]
@reportaman I have included some mouse support lately. Did you check it out?

**Comment by reportaman** [comment]
@andmarti1424 Yes! It's not reliable though... I have a big display (27inch, 5k). There are 3 problems I see:

1. On some terminal emulators, clicking beyond a certain column visible on screen sends clicks to that column instead of the column user clicked on. For WezTerm its column J. 
2. On some terminal emulators, clicking beyond a certain column does not do anything. For Alacritty & Kitty its column V.
3. On all terminal emulators, after scrolling a few pages to the right, and clicking on a cell actually sends click to some column on the left most page. Ex: Clicking on BQ8, selects a cell J8 & takes the view to J8.
4. On some terminal emulators like iTerm2 & Apple's default Terminal.app, pressing on the gutter to the right hangs sc-im, and its CPU usage remains at 100%. This happens when font size is relatively small (& column AD is visible in the opening view of sc-im), and does not happen for smaller font sizes.

**Comment by andmarti1424** [comment]
Will check point 3. 1 and 2 acting differently depending on the terminal, could be more of an ncurses issue. 4.will try to test it reducing font size in my terminal. Don't have iterm to test.

**Comment by andmarti1424** [comment]
Confirmed. number 1 and 2 seems a bug in ncurses.. event.x gives wrong position!
About number 3 you can update and test again.
Idem about number 4. I dont have how to test it.
Thanks!

**Comment by reportaman** [comment]
@andmarti1424 With your fix for 3, I confirm that both 3 & 4 too got fixed! So no worries if you don't have access to iTerm2.

With this fix there is no problem with Apple's Terminal & iTerm2, as 1 & 2 were not a problem with them to begin with. So Apple users can confidently use mouse with sc-im on Apple's Terminal & iTerm2. 

It's weird that ncurses doesn't work well with other terminals. I have been gravitating towards WezTerm, as it is the best of all IMO (at least nightly build with some recent fixes). I wonder if its those terminal emulator's bugs?

**Comment by andmarti1424** [comment]
@reportaman check this out. `https://stackoverflow.com/questions/50851532/mouse-support-with-ncurses-for-python-for-large-terminals`
this is what we see here. pass 222 columns it fails.

**Comment by andmarti1424** [comment]
but it fails sometimes. if i use alacritty + tmux it fails. if i use alacritty directly it doesnt. this is clearly a terminal thing.
